tests/test_user.py

```diff
@@ -0,0 +1,64 @@
+from .test_setup import TestSetUp
+from cp2.resources.models import Users
+from cp2.resources.api import app, db
+import json
+
+class TestRegister(TestSetUp):
+    def test_register_success(self):
+        response = self.app.post(
+            '/auth/register',
+            content_type="application/json",
+            data=json.dumps({"username": self.fakes.user_name(), "password": self.fakes.password()})
+            )
+        self.assertEqual(response.status_code, 201)
+
+    def test_register_fails_without_credentials(self):
+        response = self.app.post(
+            '/auth/register',
+            content_type="application/json",
+            data=json.dumps({"username":self.username})
+            )
+        self.assertEqual(response.status_code, 400)
+
+
+    def test_register_fails_if_user_exists(self):
+        response = self.app.post(
+            '/auth/register',
+            content_type="application/json",
+            data=json.dumps({"username":self.username, "password":self.password})
+            )
+        self.assertEqual(response.status_code, 400)
```

Comment:
Just a suggestion: 
- You can also test that the username and password are not the same... To prevent a person registering for example with 
`{
    "username": "user",
    "password": "user"
}`